src/test/java/org/assertj/core/api/junit/jupiter/SoftAssertionsExtensionAPIIntegrationTest.java

```diff
@@ -0,0 +1,152 @@
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.error.AssertJMultipleFailuresError;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.platform.testkit.engine.EngineTestKit;
+
+/**
+ * Integration tests for the public API functions of {@link SoftAssertionsExtension}.
+ *
+ * @author Fr Jeremy Krieg
+ * @since 3.18
+ */
+class SoftAssertionsExtensionAPIIntegrationTest {
+
+  static class ExtensionInjector implements ParameterResolver {
+
+    @Override
+    public boolean supportsParameter(ParameterContext parameterContext,
+                                     ExtensionContext extensionContext) throws ParameterResolutionException {
+      return parameterContext.getParameter().getType() == ExtensionContext.class;
+    }
+
+    @Override
+    public Object resolveParameter(ParameterContext parameterContext,
+                                   ExtensionContext extensionContext) throws ParameterResolutionException {
+      return extensionContext;
+    }
+
+  }
+
+  @Disabled("Executed via the JUnit Platform Test Kit")
+  @ExtendWith(ExtensionInjector.class)
+  @ExtendWith(SoftAssertionsExtension.class)
+  static class APITest {
+
+    static Map<String, AssertionErrorCollector> map = new HashMap<>();
+
+    @BeforeAll
+    static void beforeAll() {
+      map.clear();
+    }
+
+    @BeforeEach
+    void beforeEach(ExtensionContext context) {
+      SoftAssertions provider = SoftAssertionsExtension.getSoftAssertionsProvider(context, SoftAssertions.class);
+      assertThat(provider.assertionErrorsCollected()).isEmpty();
+      provider.assertThat("something").isEqualTo("nothing");
+      assertThat(provider.assertionErrorsCollected()).as("beforeEach:after assert").hasSize(1);
+      AssertionErrorCollector collector = SoftAssertionsExtension.getAssertionErrorCollector(context);
+      assertThat(provider.getDelegate()).contains(collector);
+      map.put(context.getTestMethod().get().getName(), collector);
+    }
+
+    @Test
+    void multipleFailuresCustom(ExtensionContext context, CustomSoftAssertions softly) {
+      AssertionErrorCollector collector = SoftAssertionsExtension.getAssertionErrorCollector(context);
+      assertThat(collector.assertionErrorsCollected()).as("init").hasSize(1);
+      softly.expectThat(1).isEqualTo(0);
+      assertThat(collector.assertionErrorsCollected()).as("after first").hasSize(2);
+      SoftAssertions provider = SoftAssertionsExtension.getSoftAssertionsProvider(context, SoftAssertions.class);
+      provider.assertThat(2).isEqualTo(2);
```

Comment:
I suggest to add a failing assertion for provider to make sure provider errors are collected (there is one in beforeEach but it does not hurt to add one here)